src/Hyperwallet/Util/ApiClient.php

```diff
@@ -185,7 +185,8 @@ private function doRequest($method, $url, array $urlParams, array $options) {
             )));
             throw new HyperwalletApiException($errorResponse, $e);
         } catch (BadResponseException $e) {
-            $body = \GuzzleHttp\json_decode($e->getResponse()->getBody(), true);
+            $body = $this->isEncrypted ? \GuzzleHttp\json_decode(\GuzzleHttp\json_encode($this->encryption->decrypt($e->getResponse()->getBody())), true) :
```

Comment:
- Could you split each function call into single execution that will improve readability.  or few free to introduce new function to handle the JSON decode. 

- Can you also introduce a Unit test to validate this use case as well.

Reply:
Ohh, I checked the whole method and I see that is the same approach on line 175
https://github.com/hyperwallet/php-sdk/blob/a8cccc7968c522435abd142be703022b88e88524/src/Hyperwallet/Util/ApiClient.php#L175

- Could you introduce a function to handle the body decoding ?

Reply:
@grmeyer-hw-dev made the decryption of the response a separate function and unit test added. Please check.